streaming/base/batching/per_stream.py

```diff
@@ -63,7 +63,9 @@ def generate_work_per_stream_batching(dataset: StreamingDataset, world: World, e
             # same as the ratio of the stream's samples to overall samples.
             # This ensures that the overall training shuffle block size is still approximately
             # equal to what is set by the user, and allows for reasoning about cache_limit as well.
-            assert isinstance(dataset.shuffle_block_size, int)
+            if not isinstance(dataset.shuffle_block_size, int):
+                raise TypeError(f'Dataset `shuffle_block_size` must be an integer. ' +
+                                f'Got {dataset.shuffle_block_size} instead.')
```

Comment:
nit: missed `type()`? For example: `Got {type(dataset.shuffle_block_size)} instead.`